packages/shell-api/src/collection.ts

```diff
@@ -2115,12 +2138,35 @@ export default class Collection extends ShellApiWithMongoClass {
       avgObjSize: number;
     }[] = [];
 
+    const ns = `${this._database._name}.${this._name}`;
+    let configCollectionsInfo: Document;
+    const existingConfigCollectionsInfo = await config
+      .getCollection('collections')
+      .findOne({
+        _id: ns,
+        ...onlyShardedCollectionsInConfigFilter,
+      });
+
+    if (!existingConfigCollectionsInfo) {
+      const timeseriesCollectionInfo =
+        await this._getShardedTimeseriesCollectionInfo(config, collStats);
+
+      if (!timeseriesCollectionInfo) {
+        throw new MongoshInvalidInputError(
```

Comment:
there could be more things here that have gone wrong, so one could differentiate more errors potentially